design/ceph/node-loss-rbd-rwo.md

```diff
@@ -0,0 +1,68 @@
+# Recover RBD RWO Pod in case of Node Loss
+
+## Goal
+
+Faster RBD RWO recovery in case of node loss.
+
+## Problem
+
+When a  node is lost where a pod is running with the RBD RWO volume is mounted. The volume cannot be mounted on another node since it could cause corruption if two clients are writing to the same volume.
+
+## Current Solution
+
+We have a manual solution to the problem which involves forceful deletion of a pod so that forced detachment and attachment work is possible. The problem with the current solution is that it takes around 11min to recover and also there are still chances of data corruption when the old pod on the lost node comes back online, this can cause multiple writers and lead to data corruption.
+
+## Proposed Solution
+
+> Note: This solution requires minimum kubernetes version 1.26.0
+
+We will use kubernetes beta feature [Non-Graceful Node Shutdown](https://kubernetes.io/blog/2022/12/16/kubernetes-1-26-non-graceful-node-shutdown-beta/) which will handle the volume detachment and attachment to the new node and deletion of the older pod on the lost node. To trigger this event for kubernetes we need to apply the taint  `out-of-service` manually.
+
+> Make sure to remove the taint `out-of-service` once the node is back in a good state.
+
+Once this taint is applied manually, rook will create [NetworkFencr CR](https://github.com/csi-addons/kubernetes-csi-addons/blob/main/docs/networkfence.md) using [csi-addons operator](https://github.com/csi-addons/kubernetes-csi-addons) which prevent ceph rbd client from communicating with lost node.
```

Comment:
```suggestion
Once this taint is applied manually, rook will create a [NetworkFence CR](https://github.com/csi-addons/kubernetes-csi-addons/blob/main/docs/networkfence.md). The [csi-addons operator](https://github.com/csi-addons/kubernetes-csi-addons) will then blocklist the node to prevent any ceph rbd client on the lost node from writing any more data.
```